Fix typedoc generation

Typedoc generation fails on Travis and causes the master build to fail:

```
7.21s$ if [[ "$TRAVIS_BRANCH" == "master" ]]; then lerna run typedoc && gh-pages -d docs; fi
lerna notice cli v3.4.3
lerna info Executing command in 5 packages: "npm run typedoc"
lerna ERR! npm run typedoc exited 4 in '@sakuli/commons'
lerna ERR! npm run typedoc stdout:
> @sakuli/commons@2.0.0-RC1 typedoc /home/travis/build/sakuli/sakuli/packages/sakuli-commons
> typedoc --out ../../docs/api/sakuli-cli --hideGenerator --mode file --excludeExternals --readme README.md --exclude *.spec.* src/
Using TypeScript 3.1.6 from /home/travis/build/sakuli/sakuli/packages/sakuli-commons/node_modules/typedoc/node_modules/typescript/lib
Error: /home/travis/build/sakuli/sakuli/packages/sakuli-commons/node_modules/memfs/lib/Stats.d.ts(1)
 Cannot find name 'bigint'.
Error: /home/travis/build/sakuli/sakuli/packages/sakuli-commons/node_modules/memfs/lib/Stats.d.ts(7)
 Cannot find name 'bigint'.
Error: /home/travis/build/sakuli/sakuli/packages/sakuli-commons/node_modules/memfs/lib/volume.d.ts(226)
 Cannot find name 'bigint'.
Error: /home/travis/build/sakuli/sakuli/packages/sakuli-commons/node_modules/memfs/lib/volume.d.ts(236)
 Cannot find name 'bigint'.
Error: /home/travis/build/sakuli/sakuli/packages/sakuli-commons/node_modules/memfs/lib/volume.d.ts(246)
 Cannot find name 'bigint'.
lerna ERR! npm run typedoc stderr:
npm ERR! code ELIFECYCLE
npm ERR! errno 4
npm ERR! @sakuli/commons@2.0.0-RC1 typedoc: `typedoc --out ../../docs/api/sakuli-cli --hideGenerator --mode file --excludeExternals --readme README.md --exclude *.spec.* src/`
npm ERR! Exit status 4
npm ERR! 
npm ERR! Failed at the @sakuli/commons@2.0.0-RC1 typedoc script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2019-04-12T06_35_36_713Z-debug.log
lerna ERR! npm run typedoc exited 4 in '@sakuli/commons'
The command "if [[ "$TRAVIS_BRANCH" == "master" ]]; then lerna run typedoc && gh-pages -d docs; fi" exited with 4.
```

--- a/packages/sakuli-cli/package.json
+++ b/packages/sakuli-cli/package.json
@@ -7,7 +7,7 @@
   "scripts": {
     "test": "jest",
     "build": "tsc",
-    "typedoc": "typedoc --out ../../docs/api/sakuli-cli --hideGenerator --mode file --excludeExternals --readme README.md --exclude *.spec.* src/",
+    "typedoc": "typedoc --out ../../docs/api/sakuli-cli --hideGenerator --mode file --excludeExternals --readme README.md --exclude *.spec.* --includeDeclarations src/",
     "test:watch": "jest --watch --coverage",
     "test:ci": "jest --runInBand --no-cache --coverage --passWithNoTests",
     "prepublishOnly": "npm run build && npm test"

--- a/packages/sakuli-cli/tsconfig.json
+++ b/packages/sakuli-cli/tsconfig.json
@@ -5,6 +5,7 @@
     "rootDir": "./src",
     "outDir": "./dist",
     "jsxFactory": "h",
-    "jsx": "react"
+    "jsx": "react",
+    "typeRoots": ["./node_modules/@types", "./types"]
   }
 }

--- a/packages/sakuli-commons/package.json
+++ b/packages/sakuli-commons/package.json
@@ -7,7 +7,7 @@
   "scripts": {
     "test": "jest",
     "build": "tsc",
-    "typedoc": "typedoc --out ../../docs/api/sakuli-cli --hideGenerator --mode file --excludeExternals --readme README.md --exclude *.spec.* src/",
+    "typedoc": "typedoc --out ../../docs/api/sakuli-commons --hideGenerator --mode file --excludeExternals --readme README.md --exclude *.spec.* src/",
     "test:watch": "jest --watch --coverage",
     "test:ci": "jest --runInBand --no-cache --coverage --passWithNoTests",
     "prepublishOnly": "npm run build && npm test"

--- a/packages/sakuli-core/package.json
+++ b/packages/sakuli-core/package.json
@@ -7,7 +7,7 @@
   "scripts": {
     "test": "jest",
     "build": "tsc",
-    "typedoc": "typedoc --out ../../docs/api/sakuli-cli --hideGenerator --mode file --excludeExternals --readme README.md --exclude *.spec.* src/",
+    "typedoc": "typedoc --out ../../docs/api/sakuli-core --hideGenerator --mode file --excludeExternals --readme README.md --exclude *.spec.* src/",
     "test:watch": "jest --watch --coverage",
     "test:ci": "jest --runInBand --no-cache --coverage --passWithNoTests",
     "prepublishOnly": "npm run build && npm test"

--- a/packages/sakuli-legacy/package.json
+++ b/packages/sakuli-legacy/package.json
@@ -7,7 +7,7 @@
   "scripts": {
     "test": "jest",
     "build": "tsc",
-    "typedoc": "typedoc --out ../../docs/api/sakuli-cli --hideGenerator --mode file --excludeExternals --readme README.md --exclude *.spec.* src/",
+    "typedoc": "typedoc --out ../../docs/api/sakuli-legacy --hideGenerator --mode file --excludeExternals --readme README.md --exclude *.spec.* src/",
     "test:watch": "jest --watch --coverage",
     "test:ci": "jest --runInBand --no-cache --coverage --passWithNoTests",
     "prepublishOnly": "npm run build"

--- a/packages/sakuli-legacy/src/context/sahi/relations/relations-resolver.class.spec.ts
+++ b/packages/sakuli-legacy/src/context/sahi/relations/relations-resolver.class.spec.ts
@@ -1,8 +1,16 @@
-import {By, ThenableWebDriver} from "selenium-webdriver";
+import {By, ThenableWebDriver, WebElement} from "selenium-webdriver";
 import {mockPartial} from "sneer";
 import {TestExecutionContext} from "@sakuli/core";
 import {RelationsResolver} from "./relations-resolver.class";
 import {createTestEnv, mockHtml, TestEnvironment} from "../__mocks__";
+
+const webElementToQuery = (elements: WebElement[]) => {
+    return ({
+        locator: By.js(`return arguments[0]`, elements),
+        relations: [],
+        identifier: RegExp('.*')
+    });
+};
 
 jest.setTimeout(15_000);
 describe('RelationResolver', () => {
@@ -16,7 +24,7 @@ describe('RelationResolver', () => {
 
     afterAll(async () => {
         await env.stop();
-    })
+    });
 
     function createApi(driver: ThenableWebDriver) {
         return new RelationsResolver(driver, testExecutionContext);
@@ -36,7 +44,7 @@ describe('RelationResolver', () => {
         const api = createApi(driver);
         const items = await driver.findElements(By.css('li'));
         const otherItems = items.filter((_, i) => i % 2 === 0);
-        const relationsMock = jest.fn(() => otherItems);
+        const relationsMock = jest.fn(() => Promise.resolve(webElementToQuery(otherItems)));
         const relationsMock2 = jest.fn();
         await api.applyRelations({
             relations: [
@@ -49,5 +57,4 @@ describe('RelationResolver', () => {
         expect(relationsMock).toHaveBeenCalled();
         expect(relationsMock2).toHaveBeenCalled();
     });
-
 });

--- a/packages/sakuli-rollup-hooks/package.json
+++ b/packages/sakuli-rollup-hooks/package.json
@@ -25,7 +25,7 @@
   "scripts": {
     "test": "jest",
     "build": "tsc",
-    "typedoc": "typedoc --out ../../docs/api/sakuli-cli --hideGenerator --mode file --excludeExternals --readme README.md --exclude *.spec.* src/",
+    "typedoc": "typedoc --out ../../docs/api/sakuli-rollup-hooks --hideGenerator --mode file --excludeExternals --readme README.md --exclude *.spec.* src/",
     "test:watch": "jest --watch --coverage",
     "test:ci": "jest --runInBand --no-cache --coverage --passWithNoTests",
     "prepublishOnly": "npm run build && npm test"

